src/osp/tasks/top_utils.h

```diff
@@ -57,6 +62,13 @@ namespace osp
     return current;
 }
 
+/**
+ * @brief Reserves n values in topData, where n = destLast - destFirst
```

Comment:
You're getting to deep into the implementation details.

Something like:

Reserves slots for each item in the output range [destFirst, destLast)